### PR TITLE
ws beast

### PR DESCRIPTION
**Pull request by turuslan**
- replace custom ws parsers with boost beast.
- convert all `std::error_code` to single error, returned from `libp2p::connection::LayerConnection` to boost asio.

### CODE REVIEW

File: include/libp2p/common/asio_buffer.hpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include <boost/asio/buffer.hpp>
+#include <libp2p/common/span_size.hpp>
+
+namespace libp2p {
+  inline boost::asio::const_buffer asioBuffer(
+      const gsl::span<const uint8_t> &s) {
+    return {s.data(), spanSize(s)};
+  }
+
+  inline boost::asio::mutable_buffer asioBuffer(const gsl::span<uint8_t> &s) {
```
Comment (xDimon) [inline include/libp2p/common/asio_buffer.hpp:18]: I think is better call with copy lightweight span instead use ref (aka pointer inside). Previous reloading would be enough.

File: include/libp2p/common/bytestr.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#ifndef LIBP2P_COMMON_BYTESTR_HPP
+#define LIBP2P_COMMON_BYTESTR_HPP
+
+#include <cstdint>
+#include <gsl/span>
+
+namespace libp2p {
+  inline gsl::span<const uint8_t> bytestr(const gsl::span<const char> &s) {
+    return {(const uint8_t *)s.data(), s.size()};
```
Comment (xDimon) [inline include/libp2p/common/bytestr.hpp:14]: Please, use reinterpret_cast to present intention explicitly

File: include/libp2p/layer/websocket/ws_connection.hpp [hunk 3]
```diff
@@ -6,26 +6,31 @@
 #ifndef LIBP2P_CONNECTION_WSCONNECTION
 #define LIBP2P_CONNECTION_WSCONNECTION
 
+#include <boost/beast/websocket.hpp>
```
Comment (xDimon) [inline include/libp2p/layer/websocket/ws_connection.hpp:9]: We separate internal includes, std libs includes and external
```suggestion
#include <boost/beast/websocket.hpp>

```

File: src/layer/websocket/ws_connection.cpp [hunk 4]
```diff
@@ -36,31 +37,42 @@ namespace libp2p::connection {
...
 
               // Start timer of pong waiting
               self->ping_timeout_handle_ =
                   self->scheduler_->scheduleWithHandle([wp] {
                     if (auto self = wp.lock()) {
-                      self->ws_read_writer_->close(
-                          websocket::WsReadWriter::ReasonOfClose::
-                              PINGING_TIMEOUT);
+                      self->ws_.async_close(
+                          {
```
Comment (xDimon) [inline src/layer/websocket/ws_connection.cpp:71]: Please, apply clang-format

Comment (turuslan) [reply src/layer/websocket/ws_connection.cpp:71]: It's result from both `clang-format-12` and `clang-format-15`

File: include/libp2p/common/asio_buffer.hpp (hunk 1 above)
Comment (turuslan) [reply include/libp2p/common/asio_buffer.hpp:18]: Ok, will remove reference.

No, two overloads for `const_buffer` and `mutable_buffer` are required

**Review by xDimon — APPROVED** [review]

**Pull request merged**